pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -162,6 +164,12 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 				if err := ssn.Allocate(task, node.Name); err != nil {
 					glog.Errorf("Failed to bind Task %v on %v in Session %v, err: %v",
 						task.UID, node.Name, ssn.UID, err)
+				} else {
+					// store the allocate operation on task for rollback in the future
+					if err := stmt.Allocate(task, "allocate"); err != nil {
```

Comment:
let's put all related logic into `stmt.Allocate` and `stmt.Commit` when job is ready.

Reply:
okay, it make sense.